caddyhttp/httpserver/replacer.go

```diff
@@ -132,6 +147,40 @@ func NewReplacer(r *http.Request, rr *ResponseRecorder, emptyValue string) Repla
 	return rep
 }
 
+func canLogRequest(r *http.Request) (canLog bool) {
```

Comment:
These are stylistic comments, I hope you don't mind :)

I tend to avoid named returns in these situations. I use them only rarely because they make the return statements a little more confusing, since their value lacks context. Better in this situation: leave this return value just `bool`.